src/aspire/source/__init__.py

```diff
@@ -589,12 +590,14 @@ def save_metadata(
             axis=1,
         )
 
-        if save_images:
-            # Create a new column that we will be populating in the loop below
-            # For
-            df["_rlnImageName"] = ""
+        mrcs_filenames = None
+
+        with open(starfile_filepath, "w") as f:
+            if save_images:
```

Comment:
So what happens if `save_images` is `False`? Then `mrcs_filenames` is `None`. That's not what we want. `mrcs_filenames` should always be the list of filenames and indices that were used in the STAR file.

Reply:
Changed to generate `filename_indices` from `df` no matter saving images or not in commit f16d4d2.

Reply:
Now it always uses the existing filenames. It gets overwritten below. It should be in the else clause.

Reply:
When `new_mrcs` is True, it will be updated the `df["_rlnImageName"]` column otherwise not. The filename_indices is obtained from this column. Will check it again.

Reply:
Oh ok I see. Never mind.